rclcpp/src/rclcpp/executors/multi_threaded_executor.cpp

```diff
@@ -34,9 +35,13 @@ MultiThreadedExecutor::MultiThreadedExecutor(
   yield_before_execute_(yield_before_execute),
   next_exec_timeout_(next_exec_timeout)
 {
-  number_of_threads_ = number_of_threads ? number_of_threads : std::thread::hardware_concurrency();
-  if (number_of_threads_ == 0) {
-    number_of_threads_ = 1;
+  number_of_threads_ = number_of_threads
```

Comment:
Minor: `number_of_threads` is a `size_t`, not a boolean, so prefer doing an explicit check with `number_of_threads > 0`